### PR TITLE
Groups

### PR DESCRIPTION
**Pull request by NickTomlin**
@crookedneighbor @lejeunerenard 
This adds a method to get all groups and some very light helpers for reading and writing fixtures. I added an slimmed-down, anonymized fixture for the all groups response since the real one was _gigantic_. 

I wrote a really simple "anonymizer" that recurses through objects and swaps out sensitive keys that i'll share with you all privately, it may be interesting to add it to `bin` for easier fixture creation in the future.

### CODE REVIEW

**Comment by crookedneighbor** [comment]
:candy:

**Comment by NickTomlin** [comment]
I've upped this to allow `include_participants: false` to be passed to groups all as a query (since that is likely how we will use it). The nock mocking stuff is getting verbose, perhaps we can create a helper that removes some of the boilerplate.

**Comment by crookedneighbor** [comment]
I think a test helper is a good idea, but let's make it a separate PR.

File: test/groups.test.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import {readFixture} from './support/fixture-helper';
+import Connection from '../src/connection';
+import Groups from '../src/groups';
+
+describe('Groups', () => {
+  let groups, groupsWithoutParticipants;
+  let fixture = readFixture('groups-response.xml');
+  let withoutParticipantsFixture = readFixture('groups-response-without-participants.xml');
```
Comment (lejeunerenard) [inline test/groups.test.js]: `withoutParticipantsFixture` is never used

**Comment by lejeunerenard** [comment]
besides the unnecessary `withoutParticipantsFixture`, I'm :candy:

**Pull request merged**